### PR TITLE
Updated select component (documentation and variants)

### PR DESCRIPTION
**Pull request by armstrongb**
### Context of this PR

- moved `select` component from input
- added variants for mandatory, optional and wide
- added readme file describing its implementation
- `fractal.js` updated to produce a more legible collated view of components.

### How to test

- Pull this branch and use `yarn start` to run local build.
- Navigate to the `select` component (`/components/detail/select`).
- Review the variants in preview and the documentation in the ‘Notes’ tab.

### CODE REVIEW

File: components/02-form-elements/select/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#### Mandatory
+By adding the `disabled` attribute to the ‘Select an option’ `option` the `select` element is made mandatory as it is not possible to select an option with no value.
+
+#### Optional
+Without the `disabled` attribute on ‘Select an option’ users are able to select an option without a value.
+
+### Wide
+By applying the `.input--block` class to the `select` element a variant which uses 100% of the available width can be implemented.
+
+## Context
```
Comment (shahi645) [inline components/02-form-elements/select/README.md]: If there is no context should we just get rid of this?

**Comment by shahi645** [comment]
@armstrongb am i right in thinking we shouldn't see a difference on mobile devices for the select (normal) and select (wide)?

**Comment by armstrongb** [comment]
@shahi645 yes the mobile viewport versions will become 100% width - this is expected behaviour.

File: components/02-form-elements/select/README.md (hunk 1 above)
Comment (armstrongb) [reply components/02-form-elements/select/README.md]: I was leaving that in as a placeholder for content to follow.

**Pull request merged**